test/tests/functional/pbs_reservations.py

```diff
@@ -42,6 +42,226 @@
 
 class TestReservations(TestFunctional):
 
+    """
+    Various tests to verify behavior of PBS scheduler in handling
+    reservations
+    """
+
+    def create_nodes(self):
+
+        self.server.add_resource("color", "string")
+        a = {'resources_available.ncpus': 2}
+        self.server.create_vnodes('vn', a, 2, self.mom,
+                                  usenatvnode=True, sharednode=False)
+
+        a = {'node_group_key': 'color', 'node_group_enable': 'True'}
+        self.server.manager(MGR_CMD_SET, SERVER, a)
+
+        nodes = self.server.status(NODE)
+        a = {'resources_available.color': 'blue'}
+        self.server.manager(MGR_CMD_SET, NODE, a, nodes[1]['id'])
+        self.__class__._nodes_configured = True
+
+    def submit_resv(self):
+        r = Reservation(TEST_USER)
+        a = {'Resource_List.select': '2:ncpus=1'}
+        r.set_attributes(a)
+        rid = self.server.submit(r)
+        try:
+            a = {'reserve_state': (MATCH_RE, 'RESV_CONFIRMED|2')}
+            d = self.server.expect(RESV, a, rid)
+        except PtlExpectError, e:
+            d = e.rv
+        return d
+
+    def test_no_span_psets(self):
+        self.create_nodes()
+        a = {'do_not_span_psets': '1'}
+        self.server.manager(MGR_CMD_SET, SCHED, a, "")
+        d = self.submit_resv()
+        self.assertTrue(d)
+
+    def test_over_limit(self):
+        self.create_nodes()
```

Comment:
I don't understand the purpose of this test.  Is it testing the fact that user limits do not affect reservation confirmation?